### PR TITLE
[Backend] - Delete User API

### PR DESCRIPTION
**Pull request by brentmillen**
## Ticket/s

- [[API USER] User Deletion](https://trello.com/c/E3J0TOTV/10-api-user-user-deletion)

## Changes

- Modify users.controller.ts
- Modify users.service.ts

## Screenshot

Postman
image

## How to test?

- `pnpm run dev`
- Open in Postman, import cURL to create test user
 ```
curl --location 'localhost:3001/user/create' \
--header 'Content-Type: application/json' \
--data-raw '{   
    "userType": "consumer",
    "username": "ynotony",
    "email": "iloveyou3000@email.com",
    "password": "TmOpyt820!",
    "firstName": "Tony",
    "lastName": "Start",
    "address": {
        "street": "Cornelia St.",
        "city": "Makati",
        "region": "Metro Manila",
        "zipcode": "1738",
        "country": "Philippines"
    },
    "phoneNumber": "+639292929292"
}'
```
- Copy ID of user
- import cURL
```
curl --location --globoff --request DELETE 'localhost:3001/user/{insert user ID}' \
--header 'Content-Type: application/json' \
--data ''
```
- Replace the placeholder with the ID and proceed

### CODE REVIEW

File: apps/api/src/users/users.service.ts [hunk 1]
```diff
@@ -43,16 +43,39 @@ export class UsersService {
...
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    console.log(user);
+
+    return user;
+  }
+
+  async delete(id: string) {
```
Comment (sandrex-SGV) [inline apps/api/src/users/users.service.ts]:
```suggestion
  async delete(id: string) {
  
      if (!user) {
      throw new NotFoundException('User not found');
    }
    
    
```

**Review by sandrex-SGV — CHANGES_REQUESTED** [review]
hello sir, pls add error handling that checks if user exists thanks

File: apps/api/src/users/users.service.ts (hunk 1 above)
Comment (brentmillen) [reply apps/api/src/users/users.service.ts]: done, please review thanks

**Review by sandrex-SGV — APPROVED** [review]
LGTM!

**Pull request closed without merging**